### PR TITLE
packages/libcst: add new recipe & test

### PR DESCRIPTION
**Pull request by zsol**
### Description
This PR adds a recipe and an associated test for the [libcst](https://pypi.org/project/libcst/) package.

### Checklists

- [x] Add a [CHANGELOG](https://github.com/pyodide/pyodide/blob/main/docs/project/changelog.md) entry
- [x] Add / update tests

### CODE REVIEW

File: packages/libcst/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+about:
+  home: ""
+  PyPI: https://pypi.org/project/libcst
+  summary:
+    A concrete syntax tree with AST-like properties for Python 3.0 through
+    3.12 programs.
+  license:
+    "All contributions towards LibCST are MIT licensed.  Some Python files
+    have been derived from the standard library and are therefore PSF licensed. Modifications
+    on these files are dual licensed (both MIT and PSF). These files are:  - libcst/_parser/base_parser.py
```
Comment (hoodmane) [inline packages/libcst/meta.yaml:18]: Could you cut it off after `dual licensed`?

**Review by hoodmane — APPROVED** [review]
Thanks @zsol, looks like a nice package.

File: packages/libcst/meta.yaml (hunk 1 above)
Comment (zsol) [reply packages/libcst/meta.yaml]: I removed some of the details while keeping the gist of it.

Comment (hoodmane) [reply packages/libcst/meta.yaml]: Thanks.

**Pull request merged**